MiniApp/Classes/MiniAppClient.swift

```diff
@@ -3,21 +3,27 @@ protocol SessionProtocol {
         with request: URLRequest,
         completionHandler: @escaping (Result<ResponseData, Error>) -> Void
     )
+    func startDownloadTask(downloadUrl: URL)
 }
 
-class MiniAppClient {
-    let session: SessionProtocol
+class MiniAppClient: NSObject, URLSessionDownloadDelegate {
+
     let listingApi: ListingApi
     let manifestApi: ManifestApi
+    let downloadApi: DownloadApi
     let environment: Environment
 
-    init(session: SessionProtocol = URLSession(configuration: URLSessionConfiguration.default)) {
-        self.session = session
+    override init() {
         self.environment = Environment()
         self.listingApi = ListingApi(environment: self.environment)
         self.manifestApi = ManifestApi(environment: self.environment)
+        self.downloadApi = DownloadApi(environment: self.environment)
```

Comment:
Is this where you're doing dependency setup? Seems odd to do it in the client.

Reply:
These classes were meant to create appropriate URL request for listing, manifest and downloading, which will be accessed only by the APIClient. Initially these changes were in Environment class, but later it has been moved to different classes to follow the architectural implementation of mini app